src/contents/shared/ProfileEdit.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { Component } from 'react';
+import fire from '../../config/Fire';
+import { ProfileEditView } from './ProfileEditView';
+import firebase from 'firebase';
+
+export class ProfileEdit extends Component {
+	state = {
+		  image: "https://example.com/jane-q-user/profile.jpg",
+		  url: this.props.user.photoURL,
+		  progress: 0,
+		  name: this.props.user.displayName,
+		  email: this.props.user.email,
+		  type: this.props.user.type,
+		  location: this.props.user.location,
+		  photoURL: this.props.user.url,
+		}
+	  
+	database = fire.database().ref();
+
+	/**
+		* Updates state to the current value of a certain target.
+		* @param e
+		*/
+		handleChange = e => {
+			this.setState({ [e.target.name]: e.target.value });
+		};
+
+	/**
+		* Registers the user as photographer or company.
+		*
+		* @param e
+		*/
+
+	updateUser = (e) => {
+		e.preventDefault();
+		let { name, email, type, location, url } = this.state;
+		const { user } = this.props;
+
+		if (name !== "" && email !== "" && location !== "" && url !== "") {
+
+		firebase.auth().currentUser.updateProfile(
+			{
+
+				displayName: name,
+				photoURL: url
+			}
+		)
+
+			.then(() => {
+
+				if(user.location !== location && location !=="") {
+					this.database.child("locations").child(user.location).child(user.type).child(user.uid).remove();
+					
+					
+					this.database.child("locations").child(location).child(user.type).child(user.uid).update({
+						displayName: name,
+						photoURL: url
+					});
+				}
+				else {
+					console.log("location was same as before so no update:" + user.location);
+
+				}
+
+			})
+
+			.then(() => {
```

Comment:
Please take a look at async await and try to implement this instead of the writing "then" all the time.